### PR TITLE
More easily extend web UI

### PR DESCRIPTION
**Pull request by solowalker27**
Addresses #1530. Includes the following additions and changes:

* `index.html` is extendable with blocks for tabs, panes, and scripts
* Args required to properly render `index.html` are retrievable via `web_ui.update_template_args()`
* Event for web UI's Reset Stats button to aid in custom data resetting
* Make some UI classes and functions more generic and more easily reusable for extended UI

### CODE REVIEW

**Comment by cyberw** [comment]
Cool stuff! 

Can you add some documentation and an example for how to extend the UI?

As I dont use the web ui and dont know the code behind very well, I'll take your word for it being sensible & working :) If something does break in the UI related to these changes, will you have some time to help fix it? Maybe you'd even be interested in being "the" UI maintainer for Locust :)

File: locust/static/locust.js [hunk 1]
```diff
@@ -40,15 +40,19 @@ $(".close_link").click(function(event) {
...
-        responseTimeChart.resize();
-        usersChart.resize();
-    }
+$("ul.tabs").tabs("div.panes > div").on("onClick", function (event) {
+    // trigger resizing of charts
+    resizeCharts();
 });
 
+var charts = []
+function resizeCharts() {
```
Comment (max-rocket-internet) [inline locust/static/locust.js:49]: Will this resize the charts when the window is resized?

Comment (solowalker27) [reply locust/static/locust.js:49]: Yes, the same as it did before—it's triggered only if you click on a different tab then click back on the tab with the charts.

**Comment by solowalker27** [comment]
Added example of extending the web UI.

I can help with the UI code where I can. I'm not exactly an expert, just hacking away at it. I needed to do this for my work and decided to pass the improvements upstream so I didn't have to maintain them in a fork. I don't know if I can sign on to be a maintainer of the web UI but I can provide feedback and review PRs if I can.

**Comment by cyberw** [comment]
Lgtm. Can you just add a reference to the example in the documentation (extending-locust.rst)

**Comment by solowalker27** [comment]
How's that?

**Pull request merged**